### PR TITLE
MTDSA:12636: Added coding details models

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+package v2.models.response.retrieveBalanceAndTransactions
+
+import play.api.libs.json.Json
+import support.UnitSpec
+import v2.fixtures.retrieveBalanceAndTransactions.CodingDetailsFixture._
+
+class CodingDetailsSpec extends UnitSpec {
+
+  "reads" should {
+    "return a valid model" when {
```
Comment (mattstephens-hmrc) [inline test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala:26]: please can this be something like "return the deserialized object" or something (it isn't a model)

File: test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      "valid JSON is supplied" in {
+        codingDetailsDownstreamDetailsJson
+          .as[CodingDetails] shouldBe codingDetailsObject
+      }
+    }
+  }
+
+  "writes" when {
+    "passed a valid model" should {
```
Comment (mattstephens-hmrc) [inline test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala:35]: can this be "passed a valid object" or something (again, not a model :slightly_smiling_face: )

File: test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala (hunk 1 above)
Comment (narendravijayarao) [reply test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala:26]: all other test scenarios are like this. So used the same convention

File: test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala (hunk 2 above)
Comment (narendravijayarao) [reply test/v2/models/response/retrieveBalanceAndTransactions/CodingDetailsSpec.scala]: Done

File: test/v2/fixtures/retrieveBalanceAndTransactions/CodingDetailsFixture.scala [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    coded = Some(coded))
+
+  val codingDetailsDownstreamDetailsJson: JsValue = Json.parse(
+    s"""
+       |{
+       |   "taxYearReturn": "${codingDetailsObject.taxYearReturn.get}",
+       |   "totalLiabilityAmount": ${codingDetailsObject.totalLiabilityAmount.get},
+       |   "taxYearCoding": "${codingDetailsObject.taxYearCoding.get}",
+       |   "coded": {
+       |      "charge": ${coded.charge.get},
```
Comment (sean-flynn-HMRC) [inline test/v2/fixtures/retrieveBalanceAndTransactions/CodingDetailsFixture.scala:39]: According to the specs, in the downstream "charge" is renamed to "amount". I think we need to change line 39 here and then add a corresponding reads/writes for the "Coded" object in the CodedDetails.scala model

**Pull request merged**